src/Microsoft.HttpRepl/Commands/SetHeaderCommand.cs

```diff
@@ -45,12 +45,17 @@ public Task ExecuteAsync(IShellState shellState, HttpState programState, ICorePa
 
         public string GetHelpDetails(IShellState shellState, HttpState programState, ICoreParseResult parseResult)
         {
-            var helpText = new StringBuilder();
-            helpText.Append("Usage: ".Bold());
-            helpText.AppendLine("set header {name} [value]");
-            helpText.AppendLine();
-            helpText.AppendLine("Sets or clears a header. When [value] is empty the header is cleared.");
-            return Description;
+            if (parseResult.Sections.Count > 1 && string.Equals(parseResult.Sections[0], Name, StringComparison.Ordinal) && string.Equals(parseResult.Sections[1], SubCommand, StringComparison.OrdinalIgnoreCase))
```

Comment:
Could we wrap this line into a common method to improve readability and reduce duplication, since it's used verbatim in at least 4 commands (and duplicated in `CanHandle` as well).

```
if(CommandParsingUtilitiesToBeGivenABetterName.ParseResultMatches(commandName: Name, subCommandName: SubCommand))
{ ... }
```

There isn't a common base to put the implementation in, so an extension method or helper may be appropriate...

Reply:
Per our discussion offline, I've made #60 to address a larger refactoring of this since it is reused often.